diff --git a/py/sea/commands.py b/py/sea/commands.py
index 610e33f..9511023 100644
--- a/py/sea/commands.py
+++ b/py/sea/commands.py
@@ -767,11 +767,16 @@ class Seahorn(sea.LimitedCmd):
                                         'Constrained Horn Clauses in SMT-LIB format',
                                         allow_extra=True)
         self.solve = solve
+        self.entry_pt = 'main'
 
     @property
     def stdout (self):
         return self.seahornCmd.stdout
 
+    @property
+    def stderr (self):
+        return self.seahornCmd.stderr
+
     def name_out_file (self, in_files, args=None, work_dir=None):
         return _remap_file_name (in_files[0], '.smt2', work_dir)
 
@@ -816,6 +821,8 @@ class Seahorn(sea.LimitedCmd):
         ap.add_argument ('--crab',
                          help='Enable Crab abstract interpreter',
                          dest='crab', default=False, action='store_true')
+        ap.add_argument ('--entry', help='Specify entry point function',
+                         dest='entry_pt', default='main')
         ap.add_argument ('--bmc',
                          help='Use BMC engine',
                          dest='bmc', default=False, action='store_true')
@@ -828,6 +835,7 @@ class Seahorn(sea.LimitedCmd):
 
         argv = list()
 
+        argv.append('-entry-point={}'.format(args.entry_pt))
         if args.bmc:
             argv.append ('--horn-bmc')
 
@@ -887,8 +895,12 @@ class Seahorn(sea.LimitedCmd):
         # pick out extra seahorn options
         argv.extend (filter (_is_seahorn_opt, extra))
 
-
-        return self.seahornCmd.run (args, argv)
+        rv = self.seahornCmd.run (args, argv)
+        sout = self.stdout
+        # A bit ugly.
+        if sout.strip().startswith("unsat") == True:
+            return 0
+        return 1 
 
 class SeahornClp(sea.LimitedCmd):
     def __init__ (self, quiet=False):
